### PR TITLE
Added no-non-arrow-functions rule

### LINKED ISSUE #165: new rule: ban anonymous (non-arrow) functions
Advantages as advertised here:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/arrow_functions

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
Disallows traditional `function () { ... }` declarations, preferring `() => { ... }` arrow lambdas.

Fixes #165.

### CODE REVIEW

File: src/rules/noNonArrowFunctionsRule.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ */
+
+import * as ts from "typescript";
+import * as Lint from "../lint";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "no-non-arrow-functions",
+        description: "Diallows traditional (non-arrow) function expressions.",
```
Comment (JKillian) [inline src/rules/noNonArrowFunctionsRule.ts]: typo, "Disallows"

File: test/rules/no-non-arrow-functions/test.ts.lint [hunk 2]
```diff
@@ -0,0 +1,23 @@
+function foo(a: any) : any { // error
```
Comment (JKillian) [inline test/rules/no-non-arrow-functions/test.ts.lint]: Don't need the `// error` comments in this file since the markup indicates that itself

**Comment by JKillian** [comment]
I wonder if `only-arrow-functions` would be a clearer name than `no-non-arrow-functions`? The double-negative in the latter worries me.

**Comment by JoshuaKGoldberg** [comment]
> only-arrow-functions

Agreed, way better.

**Comment by glen-84** [comment]
Also, isn't this rule supposed to only apply to anonymous functions? It's flagging stuff like:

```ts
function configureContainer(container: Container) {
    // ...
}
```

**Comment by adidahiya** [comment]
@glen-84 I agree, I expect this rule to allow named functions using the `function` keyword. would you mind filing a separate issue?

**Comment by glen-84** [comment]
Sure.

**Pull request merged**